packages/appstore/src/components/wallet-transfer/wallet-transfer.tsx

```diff
@@ -18,172 +19,269 @@ type TWalletTransferProps = {
 
 const Divider = () => <div className='wallet-transfer__divider' />;
 
-const WalletTransfer = observer(
-    ({ contentScrollHandler, is_wallet_name_visible, setIsWalletNameVisible }: TWalletTransferProps) => {
-        const { ui } = useStore();
-        const { is_mobile } = ui;
+const initial_demo_balance = 10000.0;
+
+const ERROR_CODES = {
+    is_demo: {
+        between_min_max: 'BetweenMinMax',
+        insufficient_fund: 'InsufficientFund',
+    },
+};
+
+const WalletTransfer = observer(({ is_wallet_name_visible, setIsWalletNameVisible }: TWalletTransferProps) => {
+    const { client, ui, traders_hub } = useStore();
+    const { setWalletModalActiveTab } = traders_hub;
+    const { is_switching } = client;
+    const { is_mobile } = ui;
+
+    const { getConfig } = useCurrencyConfig();
+
+    const {
+        active_wallet,
+        is_accounts_loading,
+        from_account,
+        to_account,
+        to_account_list,
+        transfer_accounts,
+        setFromAccount,
+        setToAccount,
+    } = useWalletTransfer();
+
+    useEffect(() => {
+        if (!from_account?.loginid) {
+            setFromAccount(active_wallet);
+        }
+    }, [active_wallet, from_account, setFromAccount]);
+
+    const portal_id = is_mobile ? 'mobile_list_modal_root' : 'modal_root';
+
+    const is_amount_to_input_disabled = !to_account;
+
+    const active_wallet_name = getAccountName({ ...active_wallet });
+
+    const transfer_to_hint = React.useMemo(() => {
+        return to_account?.loginid === active_wallet?.loginid ? (
+            <Localize
+                i18n_default_text='You can only transfers funds from the {{account}} to the linked {{wallet}}.'
+                values={{
+                    account: getAccountName({ ...from_account }),
+                    wallet: active_wallet_name,
+                }}
+            />
+        ) : (
+            ''
+        );
+    }, [active_wallet?.loginid, active_wallet_name, from_account, to_account?.loginid]);
+
+    const [message_list, setMessageList] = React.useState<TMessageItem[]>([]);
```

Comment:
Message list should only contain things that are required to render out the component! It should not contain things that determine its logic. All of this should be handled internally in the components!